File: packages/gokai/lib/services/storage_manager.dart
import 'package:gokai/fs/entry.dart';
import 'package:gokai/service.dart';

class GokaiStorageManager extends GokaiService {
  GokaiStorageManager() : super(serviceName: 'StorageManager');

  GokaiFileSystemEntry open(String path) => GokaiFileSystemEntry(path);
}

File: packages/gokai_sdk/framework/lib/os/linux/services/storage-manager.cc
#include <gokai/framework/os/linux/fs/entry.h>
#include <gokai/framework/os/linux/services/storage-manager.h>

using namespace Gokai::Framework::os::Linux::Services;

StorageManager::StorageManager(Gokai::ObjectArguments arguments) : Gokai::Services::StorageManager(arguments) {}

Gokai::FS::Entry StorageManager::open(std::filesystem::path path) {
  return Gokai::Framework::os::Linux::FS::Entry(Gokai::ObjectArguments({
    { "context", this->context },
    { "path", path },
  }));
}
